scalameta/trees/src/main/scala/scala/meta/Trees.scala

```diff
@@ -124,6 +124,7 @@ object Type {
   @ast class Singleton(ref: Term.Ref) extends Type.Ref with Pat.Type.Ref {
     require(ref.isPath || ref.is[Term.Super])
   }
+  @ast class Literal(literal: Lit) extends Type with Pat.Type
```

Comment:
Originally, we planned to use just `Lit` for this purpose. Hence, it already extends `Type`. Do you think you'd be able to get by without a dedicated AST node?